### ISSUE TITLE
Question: Schema Transform partial application of arguments

### ISSUE THREAD

**Issue by m-sanders** [yaacovCR/graphql-tools-fork#44]
Hello!

We have a remote schema with a required argument `lang` on all root fields, we would like to set this value via a transform and remove it as an argument from the exposed API (or set the default value).

We were previously on version 7.2.3 and I achieved the above by using the below `TransformRootFields` to set the `defaultValue`:

```
import { fieldToFieldConfig } from "graphql-tools-fork/dist/stitching/schemaRecreation";

// ...

new TransformRootFields((_operation, name, field) => {
  // Set any root fields with a `lang` arg to default to the correct locale.
  const config = fieldToFieldConfig(field);

  if (config.args.lang !== undefined) {
    config.args.lang.defaultValue = locale.langtag.toLowerCase();
  }

  return { name, field: config };
})

```

But on upgrading to the latest version to resolve #43 , `fieldToFieldConfig` has been removed. What's the best way to go about this without `fieldToFieldConfig`?

**Comment by yaacovCR** [comment]
Now that we have a toConfig function that polyfills graphql-js functionality, we can add field ToFieldConfig to that.

In the meantime, you can grab schema as is by saving it to private variable within a transformSchema function, and then use the field.name to get the field definition from type using existing type toConfig

**Comment by yaacovCR** [comment]
You should now be able to use toConfig or fieldToConfig.

Let me know if you have any issues.

**Comment by m-sanders** [comment]
Thanks @yaacovCR. I tried this out today and my implementation no longer works. 

When the transformed schema `delegatesToSchema` the new `defaultValue` is not sent and the remote receives `null`. As this is a non null field, I receive a GraphQLError complaining about `lang` being required. Do you have any ideas on how I might best solve this issue?

**Comment by yaacovCR** [comment]
Yes! I made a mistake within the createRequest logic, if you want you can open another issue to track, have fix in works.

**Comment by yaacovCR** [comment]
@m-sanders , can you confirm fixed in v9? Thanks!

**Comment by m-sanders** [comment]
@yaacovCR Will do ASAP. Thanks again.

**Comment by m-sanders** [comment]
@yaacovCR It looks like it is fixed in 9.0.0! Thanks!

I will do some more testing to make sure.

**Comment by m-sanders** [comment]
One unexpected change with the upgrade, previously optional and omitted resolver arguments were `undefined` and now they are `null`. I had a quick look through the Changelog but couldn't see if this is an expected change.  Is this expected behaviour?

**Comment by yaacovCR** [comment]
Hi, I found an obvious (?) bug with default values, where I was checking defaultValue != null when I should have been checking defaultValue !== undefined. This has been fixed in 9.0.1.

Can you check if this fixes your issue? Otherwise, I may need to ask you for a small reproducing example as it would definitely expedite.

**Comment by m-sanders** [comment]
Hi. That looks like it has resolved the issue. Thanks again!
